.happy/terraform/modules/ecs-stack/main.tf

```diff
@@ -217,3 +223,113 @@ module upload_sfn {
   lambda_error_handler   = module.upload_error_lambda.arn
   deployment_stage       = local.deployment_stage
 }
+
+module dataset_submissions_lambda {
+  source                     = "../lambda"
+  image                      = "${local.lambda_dataset_submissions_repo}:${local.image_tag}"
+  name                       = "dataset-submissions"
+  custom_stack_name          = local.custom_stack_name
+  remote_dev_prefix          = local.remote_dev_prefix
+  deployment_stage           = local.deployment_stage
+  artifact_bucket            = local.artifact_bucket
+  cellxgene_bucket           = local.cellxgene_bucket
+  dataset_submissions_bucket = local.dataset_submissions_bucket
+  lambda_execution_role      = aws_iam_role.dataset_submissions_lambda_service_role.arn
+  step_function_arn          = module.upload_sfn.step_function_arn
+  subnets                    = local.subnets
+  security_groups            = local.security_groups
+}
+
+resource "aws_iam_role" "dataset_submissions_lambda_service_role" {
+  name               = "corpora-dataset-submissions-service-role-${var.deployment_stage}"
+  path               = "/service-role/"
+  assume_role_policy = data.aws_iam_policy_document.assume_role.json
+}
+
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+data "aws_iam_policy_document" "lambda_step_function_execution_policy" {
+  statement {
+    sid    = "sfn"
+    effect = "Allow"
+    actions = [
+      "states:StartExecution"
+    ]
+    resources = [
+      module.upload_sfn.step_function_arn
+    ]
+  }
+  statement {
+    sid      = "cw"
+    effect = "Allow"
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents"
+    ]
+    resources = [
+      "arn:aws:logs:*:*:*"
+    ]
+  }
+  statement {
+    actions = [
+      "secretsmanager:DescribeSecret",
+      "secretsmanager:GetSecretValue"
+    ]
+    resources = [
+      "arn:aws:secretsmanager:us-west-2:${var.aws_account_id}:secret:corpora/backend/${var.deployment_stage}/*"
+    ]
+  }
+  statement {
+    sid     = "s3"
+    effect  = "Allow"
+    actions = [
+      "s3:ListBucket",
```

Comment:
`ListBucket` and `DeleteObject` are required to delete artifacts when a dataset is updated.

Reply:
Don't we also need `CreateObject`?

Reply:
The S3 upload (and `CreateObject`) to the dataset submissions bucket is done outside of this lambda by the user of the curation API uploading directly to a pre-signed URL.